swiss_locator/core/filters/swiss_locator_filter.py

```diff
@@ -498,6 +494,27 @@ def triggerResult(self, result: QgsLocatorResult):
                     QgsProject.instance().addMapLayer(ch_layer, False)
                     root.insertLayer(-1, ch_layer)
 
+                if empty_project:
+                    QCoreApplication.processEvents()  # wait for application to process pending tasks
```

Comment:
@3nids, Not completely sure of this, but seems to be necessary.
Do you know of any other way to get the `setExtent()`; `refresh()` work without this `processEvents()` call?

Reply:
Avoided.